src/CHIPBackend.hh

```diff
@@ -1536,6 +1536,21 @@ public:
    * @return CHIPContext*
    */
   CHIPContext *retain();
+
+  /**
+   * @brief Get the smallest max allocation size for this context by checking
+   * the max allocation size across all devices in this context.
+   *
+   * @return size_t
+   */
+  size_t getMaxAllocSize() {
+    size_t AllocSize = std::numeric_limits<size_t>::max();
+    for (auto Dev : getDevices()) {
```

Comment:
Why do you need to scan through all the device? I would assume current device's max limit should be enough to check.

Reply:
Because allocations take place inside a context and a context can be created for multiple devices. I assume it's possible that these devices have different properties? 

```
  // Load devices to device vector
  Status = zeDeviceGet(ZeDriver, &DeviceCount, nullptr);
  CHIPERR_CHECK_LOG_AND_THROW(Status, ZE_RESULT_SUCCESS, hipErrorTbd);
  ZeDevices.resize(DeviceCount);
  Status = zeDeviceGet(ZeDriver, &DeviceCount, ZeDevices.data());
  CHIPERR_CHECK_LOG_AND_THROW(Status, ZE_RESULT_SUCCESS, hipErrorTbd);

  const ze_context_desc_t CtxDesc = {ZE_STRUCTURE_TYPE_CONTEXT_DESC, nullptr,
                                     0};

  ze_context_handle_t ZeCtx;
  Status = zeContextCreateEx(ZeDriver, &CtxDesc, DeviceCount, ZeDevices.data(),
                             &ZeCtx);
  CHIPERR_CHECK_LOG_AND_THROW(Status, ZE_RESULT_SUCCESS, hipErrorTbd);
```